### DISCUSSION TITLE
Chatbots

### DISCUSSION THREAD

**Discussion by vjFaLk** [Coronasafe/life#146] [Show and tell]
Continuing the discussion from - 

https://github.com/coronasafe/life/issues/140

Tagging @hannanabdul55 @bodhish 

> > What are you thoughts on porting your code over to a Telegram bot.
> 
> I am not sure about the Telegram API. But I can definitely standardize the data querying here and serve that through the flask server. I am not sure about what aspects of the natural language processing does Telegram handle. If it does all the parsing into Entities, similar to Dialogflow, then it should be pretty easy to port the entire code to Telegram. As a matter of fact, [Dialogflow itself has an option](https://cloud.google.com/dialogflow/es/docs/integrations/telegram) to "plug in" to Telegram.

Haha I wish I knew about this before I hacked together a Telegram bot from scratch. However, by having a bot directly interface with the API we remove the GCP dependency and also avoid costs / limits. The message processing on my bot is very rudimentary but it works. You can test it here - 

https://t.me/dev_coronasafe_life_bot

**Warning** - Just for testing, don't share please. 

Here's the source code - https://github.com/vjFaLk/coronasafe-life-bot

Example - 

![image](https://user-images.githubusercontent.com/6697133/116206058-08b76a80-a75c-11eb-80b9-8c2c630eada9.png)

**Comment by vjFaLk** [comment]
Let me know what you guys think, I'm open to feedback, and if DialogFlow's Telegram integration is better (and the costs / limits are not a problem), then so be it, rather use that.

**Reply by vigneshhari** [reply]
You can share this in the slack workspace for better visibility, post it in the coronasafe-life channel

**Reply by vjFaLk** [reply]
Hey @vigneshhari I don't have access to the Slack, could you share an invite / link?

**Reply by vigneshhari** [reply]
http://slack.coronasafe.in/
